language/llama3-405b/dataset.py

```diff
@@ -0,0 +1,129 @@
+import random
+import os
+import time
+import numpy as np
+import torch
+from datasets import load_dataset, load_from_disk
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from torch.nn.functional import pad
+from torch.utils.data import DataLoader
+from typing import Optional, Dict, Sequence
+import io
+
+# import utils
+import copy
+import pickle
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("Llama-70B-Dataset")
+
+
+class Dataset:
+    def __init__(
+        self,
+        model_name=None,
+        total_sample_count=24576,
+        perf_count_override=None,
+        dataset_path=None,
+        dtype="bfloat16"
+    ):
+        self.model_name = model_name or f"Meta-Llama-3.1-405B-Instruct{'-FP8' if dtype == 'float8' else ''}"
+        self.dataset_path = dataset_path
+        self.max_length = 1024
```

Comment:
I forgot this is ISL or OSL max length, but it should be 20000/2000

Reply:
Removed, this was not used